### ISSUE TITLE
Train a Cd surrogate model using the C++ ML framework

### ISSUE THREAD

**Issue by MarcosAsh** [MarcosAsh/3dFluidDynamicsInC#53]
Use the existing C++ ML framework (deepseek_like_ai_c) to train a small neural network that predicts drag coefficients from mesh geometry and flow parameters. This replaces the JAX/Python approach with our own stack.

## Why

The LBM solver takes 30-60s per run on a T4 GPU. A trained surrogate model could predict Cd in microseconds, enabling instant feedback in the UI while the full render runs in the background. Keeping everything in C/C++ means no Python dependency at inference time and the model can run directly inside the simulation loop.

## Architecture

**Training (C++ framework)**

The framework already has everything needed: autodiff, linear layers, AdamW, activations. The surrogate model is a simple regression MLP:

```
Input: [voxelized_mesh (flattened occupancy grid), wind_speed, reynolds_number]
  -> Linear(input_dim, 256) -> SwiGLU
  -> Linear(256, 128) -> SwiGLU
  -> Linear(128, 2)
Output: [Cd, Cl]
```

Input representation options (try both, see what works):
1. **Voxelized occupancy grid** -- rasterize the OBJ to a small grid (e.g. 32x16x16 = 8192 binary values), concatenate with wind_speed and Re. Simple, fixed-size input.
2. **Bounding box + shape descriptors** -- frontal area, length/width/height ratios, surface area, volume. Much smaller input (~10-20 features) but less expressive.

Option 1 is more general (works for arbitrary shapes), option 2 trains faster and might be enough for the models we have.

**Data generation (Modal)**

Write a script that calls the Modal render endpoint in a loop to generate training pairs:
- Models: car, ahmed25, ahmed35 (and eventually custom OBJs)
- Wind speeds: 0.5 to 5.0 in steps of 0.25
- Reynolds numbers: auto (default), 1000, 5000, 10000, 50000
- Duration: 15s minimum to let Cd converge
- Store: (input features, final Cd, final Cl) as binary records

Target ~500-1000 training samples to start. The Modal worker already returns cd_value and cl_value so the pipeline is mostly just orchestration.

**Inference in C (inside the simulation)**

Once training is done, export weights as a flat binary file (just dump the float arrays). Write a minimal C forward pass:

```c
// ~50 lines, no dependencies
typedef struct { float* weights; float* biases; int in; int out; } Layer;

float* mlp_forward(Layer* layers, int n_layers, float* input) {
    // matrix multiply + activation for each layer
}

float predict_cd(float* voxel_grid, float wind_speed, float reynolds) {
    // pack input, run forward pass, return Cd
}
```

This runs inside the render loop and displays an estimated Cd immediately on startup, before the LBM has converged. The UI can show "Estimated Cd: 0.28 (ML) / Computing... (LBM)" and update once the real value is ready.

**Inference on the web (Modal endpoint)**

Wrap the trained model in a lightweight Modal function that takes the same parameters as the render endpoint but returns instantly. The website calls this first for an instant estimate, then kicks off the full render.

## Steps

1. Write a data generation script that batch-runs simulations on Modal and saves results
2. Write a voxelizer that converts OBJ meshes to fixed-size occupancy grids
3. Set up a training loop in the C++ framework (MLP regression, MSE loss, AdamW)
4. Train on the generated dataset, validate on held-out samples
5. Export weights to a binary format
6. Write the C inference forward pass and integrate into main.c
7. Add a Modal endpoint for web-based instant prediction
8. Wire up the website to call the prediction endpoint before rendering

## What we already have

- C++ ML framework with autodiff, MLP layers, AdamW, SwiGLU activations, weight export
- Modal GPU worker that returns Cd/Cl values
- Website that displays coefficient results
- OBJ model loader in the simulation code

## Success criteria

- Predicted Cd within 15% of the LBM-computed value for seen model geometries
- Inference time under 1ms in C, under 50ms via the Modal endpoint
- Graceful fallback if no trained weights are available (just skip the prediction)
